### PR TITLE
Adding hazelbean

### PR DESCRIPTION
**Pull request by phargogh**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by phargogh** [comment]
I confirm that I am willing to be a recipe maintainer.  @jandrewjohnson could you confirm that you are willing to be a maintainer as well?  Please comment here if so.

**Comment by jandrewjohnson** [comment]
I confirm that I am willing to be a recipe maintainer.

**Comment by phargogh** [comment]
Ok, I think this is ready for review! @conda-forge/help-python

File: recipes/hazelbean/LICENSE [hunk 1]
```diff
@@ -0,0 +1,98 @@
+https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode
```
Comment (ocefpaf) [inline recipes/hazelbean/LICENSE:1]: This file is in the tarball. You don't need to re-add it here.

File: recipes/hazelbean/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - python
+    - cython <3
+    - numpy
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:21]: You likely don't need these in build.

```suggestion
```

File: recipes/hazelbean/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    - anytree
+    - statsmodels
+    - openpyxl
+    - seaborn
+    - twine
+    - pyqt
+    - ipykernel
+    - imageio
+    - pandoc
+    - conda
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:63]: You should have conda to install a conda package!

```suggestion
```

File: recipes/hazelbean/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    - rioxarray
+    - dask
+    - google-cloud-datastore
+    - google-cloud-storage
+    - aenum
+    - anytree
+    - statsmodels
+    - openpyxl
+    - seaborn
+    - twine
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:58]: twine is a build tool. It is unlikely that you need it to run this.

```suggestion
```

**Comment by ocefpaf** [comment]
Upstream declares no dependency at all. The dependencies here are a bit odd, a mix of build and run tools into an env. Is that correct? Running depfinder I found:

```
 'required': ['Cython',
              'Pyro4',
              'aenum',
              'dask',
              'distutils',
              'fiona',
              'geopandas',
              'google.cloud',
              'hazelbean',
              'markdown',
              'matplotlib',
              'natcap',
              'netcdf4',
              'numpy',
              'osgeo',
              'pandas',
              'pygeoprocessing',
              'pyyaml',
              'qtawesome',
              'qtpy',
              'rioxarray',
              'scikit-learn',
              'scipy',
              'setuptools',
              'shapely',
              'sip',
              'six',
              'statsmodels',
              'taskgraph',
              'xarray',
              'xlrd']}
```

File: recipes/hazelbean/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - python
+    - cython <3
+    - numpy
+    - git
```
[the remark this answers is not in the document]
Comment (phargogh) [reply recipes/hazelbean/meta.yaml]: I don't remember why I had this in here, so it's been removed.

**Comment by phargogh** [comment]
@ocefpaf you are very right, the dependencies were a combination of actual dependencies and build tools, which were taken from the upstream package's documentation, but clearly this list was a bit larger than was needed for this package, specifically.  Thanks for your review!

File: recipes/hazelbean/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  host:
+    - python >=3.6
+    - pip
+    - cython <3
+    - numpy
+    - setuptools
+    - setuptools_scm
+    - {{ stdlib("c") }}
+  run:
+    - Cython
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:30]: Unless you do some code generation at run time, it is very likely that cython is not required to run this.
```suggestion
```

File: recipes/hazelbean/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    - pyqt
+    - python >=3.6
+    - pyyaml
+    - qtawesome
+    - qtpy
+    - rioxarray
+    - scikit-learn
+    - scipy
+    - seaborn
+    - setuptools
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:60]: You rarely need setuptools at run time. Are you using `pkg_resources`?

```suggestion
```

Comment (phargogh) [reply recipes/hazelbean/meta.yaml:60]: You're right, `setuptools` is only needed at build time, not at runtime.

File: recipes/hazelbean/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+  host:
+    - python >=3.6
+    - pip
+    - cython <3
+    - numpy
+    - setuptools
+    - setuptools_scm
+    - {{ stdlib("c") }}
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:28]:
```suggestion
```

File: recipes/hazelbean/meta.yaml (hunk 6 above)
Comment (ocefpaf) [reply recipes/hazelbean/meta.yaml:30]: Ping

File: recipes/hazelbean/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    - Pyro4
+    - aenum
+    - anytree
+    - dask
+    - descartes
+    - fiona
+    - geopandas
+    - google-cloud-datastore
+    - google-cloud-storage
+    - libgcc
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:40]: Installing at runtime libgcc is not advised. It is very unlikely that you need this.

```suggestion
```

File: recipes/hazelbean/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    - pyyaml
+    - qtawesome
+    - qtpy
+    - rioxarray
+    - scikit-learn
+    - scipy
+    - seaborn
+    - shapely
+    - sip
+    - six
```
Comment (ocefpaf) [inline recipes/hazelbean/meta.yaml:62]: Can you check if you really need six? Are you still supporting Python 2 syntax with it?

```suggestion
```

Comment (phargogh) [reply recipes/hazelbean/meta.yaml:62]: While I agree with you that python2 support is no longer needed, a vendored version of a library within `hazelbean` does still use `six`, so it is, at this time, still a dependency.

File: recipes/hazelbean/meta.yaml (hunk 6 above)
Comment (phargogh) [reply recipes/hazelbean/meta.yaml]: Removed in dfab217e85

File: recipes/hazelbean/meta.yaml (hunk 8 above)
Comment (phargogh) [reply recipes/hazelbean/meta.yaml]: I just removed this and also `libgcc` in cef1607586.

File: recipes/hazelbean/meta.yaml (hunk 9 above)
Comment (phargogh) [reply recipes/hazelbean/meta.yaml]: I just removed this and `stdlib(c)` in cef1607586.  The builds were failing without including `libgcc` before, so I'm curious if removing both lines would result in a successful build.

**Comment by phargogh** [comment]
@conda-forge-admin please rerender

**Comment by phargogh** [comment]
@ocefpaf I'm really hoping you can help me here with a build issue.  Both the mac and linux builds are failing because there is an unsatisfiable requirement, and, to my eyes, it looks like conda-forge is trying to build binaries for python 2.7.  I've tried everything I can think of to restrict the python versions to 3.8+ (the currently-supported conda-forge build versions), but I'm at a loss of what else to try.  Would it be possible to take a look and see what I'm missing?  Thank you in advance!

**Comment by ocefpaf** [comment]
> @ocefpaf I'm really hoping you can help me here with a build issue. Both the mac and linux builds are failing because there is an unsatisfiable requirement, and, to my eyes, it looks like conda-forge is trying to build binaries for python 2.7. I've tried everything I can think of to restrict the python versions to 3.8+ (the currently-supported conda-forge build versions), but I'm at a loss of what else to try. Would it be possible to take a look and see what I'm missing? Thank you in advance!

Looks like you figure it out, right?

PS: We dropped 2.7 a long time ago!

**Pull request merged**